### ISSUE TITLE
RFC: Effects should not be triggered when using time travel debugger

### ISSUE THREAD

**Issue by rstraub** [ngrx/platform#1062]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[X] Feature request
[ ] Documentation issue or request
</code></pre>

## What is the current behavior?
Currently when you use the effects module and you use the time travel debugger the effect is triggered when replaying. This can for instance cause http requests to trigger again. In turn this causes the state to differ between the first "recording" of the app state and after replaying since it calls the endpoint during each replay.

This is my client after clicking through the first time manually. I added a new item to my list called "Create feature request for NgRx". This makes a call to the server. The result is this:
<img width="1161" alt="screen shot 2018-05-13 at 13 36 56" src="https://user-images.githubusercontent.com/2283567/39966855-d330e028-56b2-11e8-8659-0cbf4c9cb4b0.png">

If I then replay using the time travel debugger, I end up with this:
<img width="1160" alt="screen shot 2018-05-13 at 13 38 46" src="https://user-images.githubusercontent.com/2283567/39966877-0b82f01a-56b3-11e8-874d-3cdf7266f998.png">

It is clear to see that it actually added the record again, mutating the state and possibly causing different behaviour between manually editing state and afterwards replaying it.

## Expected behavior:
I would expect the triggered effects not to be invoked during replay with the time travel debugger. This keeps the clients state consistent during replays and prevents weird behaviour dependent on the side effect model (like an http request returning an error the second time an endpoint is invoked).

My request is to ignore effects when replaying and only replay downstream actions like a success or failure which don't mutate state outside of the application's context.

**Comment by dummdidumm** [comment]
To my knowledge this is not possible because it is not clear to the app where the actions come from. But you can "ignore" all actions and state changes once #955 lands using "lock".

**Comment by brandonroberts** [comment]
Duplicate of #822
